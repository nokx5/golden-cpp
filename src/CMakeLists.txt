add_executable(cli_golden_cpp cli_golden_cpp.cpp)
# add_library(golden SHARED src/golden.cpp src/silver/silver.hpp)
# target_link_libraries(golden ${BOOST_LIBRARIES})

install(
  TARGETS cli_golden_cpp
  CONFIGURATIONS Debug
  DESTINATION debug/bin)

install(
  TARGETS cli_golden_cpp
  CONFIGURATIONS Release
  DESTINATION bin)
