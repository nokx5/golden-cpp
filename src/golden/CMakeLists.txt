set(libraries gold)

foreach(library ${libraries})

  if(${PROJECT_STATIC})
    add_library(${library} STATIC core/${library}.cpp)
  else()
    add_library(${library} SHARED core/${library}.cpp)
  endif()
  target_link_libraries(${library}) # ${BOOST_LIBRARIES} Boost::system
                                    # Boost::filesystem

endforeach()

add_executable(cli_golden_cpp cli/golden.cpp)
# target_link_libraries(golden ${BOOST_LIBRARIES})

install(
  TARGETS cli_golden_cpp
  CONFIGURATIONS Release
  DESTINATION bin)
